pkg/controller/node/node_controller.go

```diff
@@ -613,15 +613,21 @@ func (ctrl *Controller) getPoolsForNode(node *corev1.Node) ([]*mcfgv1.MachineCon
 	if len(custom) > 1 {
 		return nil, fmt.Errorf("node %s belongs to %d custom roles, cannot proceed with this Node", node.Name, len(custom))
 	} else if len(custom) == 1 {
-		// We don't support making custom pools for masters
+		pls := []*mcfgv1.MachineConfigPool{}
 		if master != nil {
-			return nil, fmt.Errorf("node %s has both master role and custom role %s", node.Name, custom[0].Name)
+			// if we have a custom pool and master, defer to master and return.
+			glog.Infof("Given both master and custom pools. Defaulting to master: custom %v", custom[0].Name)
+			ctrlcommon.MCCPoolAlert.WithLabelValues(custom[0].Name, fmt.Sprintf("Given both master and custom pools. Defaulting to master: custom %v", custom[0].Name)).Set(1)
```

Comment:
suggestion: I think we should expand this to something like:

`Found master node that matches selector for custom pool %v, defaulting to master. This node will not have any custom role configuration as a result. Please review the node to make sure this is intended`

Along those lines for clarity for the user?